src/main/java/org/opentripplanner/routing/algorithm/filterchain/filters/NonTransitGeneralizedCostFilter.java

```diff
@@ -0,0 +1,49 @@
+package org.opentripplanner.routing.algorithm.filterchain.filters;
+
+import org.opentripplanner.model.plan.Itinerary;
+import org.opentripplanner.routing.algorithm.filterchain.ItineraryFilter;
+
+import java.util.List;
+import java.util.OptionalDouble;
+import java.util.function.DoubleFunction;
+import java.util.stream.Collectors;
+
+/**
+ * This filter is similar to {@link TransitGeneralizedCostFilter}, but it will only remove
+ * non-transit results.
+ * <p>
+ * @see org.opentripplanner.routing.api.request.ItineraryFilterParameters#nonTransitGeneralizedCostLimit
```

Comment:
Describe the intention of the filter - why is it needed? Again it is nice with the cross reference, but you still need to document the filter.